src/Microsoft.Azure.Functions.Extensions.Mcp/Sessions/McpClientSessionManager.cs

```diff
@@ -24,28 +23,44 @@ private void RemoveSession(string clientId)
         logger.LogInformation("Removed session for client '{ClientId}'.", clientId);
     }
 
-    public IMcpClientSession<TTransport> CreateSession<TTransport>(string clientId, string instanceId, TTransport transport) where TTransport : ITransportWithMessageHandling
+    public ValueTask<IMcpClientSession<TTransport>> CreateSessionAsync<TTransport>(string clientId, string instanceId, TTransport transport) where TTransport : IMcpExtensionTransport
     {
-        var clientSession = new McpClientSessionImplementation<TTransport>(this, clientId, instanceId, transport);
+        IMcpClientSession<TTransport> clientSession = new McpClientSessionImplementation<TTransport>(this, clientId, instanceId, transport);
 
-        if (!_sessions.TryAdd(clientId, clientSession))
+        if (!transport.IsStateless)
         {
-            // This condition should never happen
-            throw new InvalidOperationException($"A session for client '{clientId}' already exists.");
+            if (!_sessions.TryAdd(clientId, clientSession))
```

Comment:
Stateful is supported here?

Reply:
Yes. This is what SSE will use for session management as well. Once we have stateful support for streamable, there will be additional work for proper support of distributed sessions and sessions with persistent connections (SSE for notifications, etc.).